### PR TITLE
Make the model factories more dynamic

### PR DESCRIPTION
**Pull request by jbrooksuk**
The model factories have a bit more variety to them. Tests should be more reliable.

### CODE REVIEW

File: database/factories/ModelFactory.php [hunk 1]
```diff
@@ -34,7 +34,7 @@
     return [
         'name'      => $faker->words(2, true),
         'order'     => 0,
-        'collapsed' => false,
+        'collapsed' => (bool) rand(0, 1),
```
Comment (joecohens) [inline database/factories/ModelFactory.php]: I think there is a $faker->boolean() do we prefer it here?

Comment (jbrooksuk) [inline database/factories/ModelFactory.php]: Ah, is there? We'd be better using that.

**Comment by jbrooksuk** [comment]
@joecohens fixed :)

File: database/factories/ModelFactory.php [hunk 2]
```diff
@@ -25,7 +25,7 @@
         'name'        => $faker->sentence(),
         'description' => $faker->paragraph(),
         'link'        => $faker->url(),
-        'status'      => 1,
+        'status'      => rand(1, 4),
```
Comment (GrahamCampbell) [inline database/factories/ModelFactory.php]: better to use random_int

Comment (GrahamCampbell) [inline database/factories/ModelFactory.php]: that should be used always instead of rand or mt_rand

Comment (GrahamCampbell) [inline database/factories/ModelFactory.php]: it also actually has better performance than mt_rand

File: database/factories/ModelFactory.php [hunk 3]
```diff
@@ -61,14 +61,14 @@
...
-        'display_chart' => 1,
+        'calc_type'     => $faker->boolean(),
+        'display_chart' => $faker->boolean(),
     ];
 });
 
 $factory->define(MetricPoint::class, function ($faker) {
     return [
-        'metric_id' => 1,
+        'metric_id' => factory('CachetHQ\\Cachet\\Models\\Metric')->create()->id,
```
Comment (GrahamCampbell) [inline database/factories/ModelFactory.php]: probably best to use ::class notation

File: database/factories/ModelFactory.php (hunk 2 above)
Comment (jbrooksuk) [inline database/factories/ModelFactory.php]: :+1:

**Pull request merged**